src/main/scala/fi/oph/koski/raportit/ExcelWriter.scala

```diff
@@ -173,6 +189,45 @@ object ExcelWriter {
     cell.setCellValue(f)
   }
 
+  private def createComment(drawing: SXSSFDrawing, anchor: ClientAnchor, creationHelper: CreationHelper, commentText: String) = {
+    val comment = drawing.createCellComment(anchor)
+    comment.setString(creationHelper.createRichTextString(commentText))
+    comment
+  }
+
+  private def createAnchor(creationHelper: CreationHelper, sh: SXSSFSheet, cell: Cell, commentText: String) = {
+    //Size of an Comment is reserved by allocating cells and rows
+    val leftColumnIndex = cell.getColumnIndex
+    val (rightColumnIndex, rowIndex) = calucateCommentSize(sh, leftColumnIndex, commentText)
+
+    val anchor = creationHelper.createClientAnchor
+    anchor.setRow1(HEADING_ROW)
+    anchor.setRow2(rowIndex)
+    anchor.setCol1(leftColumnIndex)
+    anchor.setCol2(rightColumnIndex)
+    anchor
+  }
+
+  private def calucateCommentSize(sh: SXSSFSheet, startIndex: Int, commentText: String) = {
+    val commentMaxWidth = 20000
+
+    val textWidth = commentText.length * ONE_COMMENT_CHARACTER_WIDTH
+
+    val requiredWidth = Math.min(commentMaxWidth, textWidth)
+
+    val requiredHeight = textWidth.toDouble / commentMaxWidth
+    val rowIndex = if (requiredHeight < 2) 2 else requiredHeight.floor.toInt
+
+    var currentWidth = 0
+    var columnIndex = startIndex
+    while (currentWidth < requiredWidth) {
+      currentWidth += sh.getColumnWidth(columnIndex)
```

Comment:
Is it possible to get index out of bounds exception from this? Or is it possible that getColumnWidth returns 0, which could lead to infinite loop?